OsmAnd/src/net/osmand/plus/views/layers/core/NearbyPlacesTileProvider.java

```diff
@@ -24,46 +16,38 @@
 import net.osmand.core.jni.TileId;
 import net.osmand.core.jni.ZoomLevel;
 import net.osmand.core.jni.interface_MapTiledCollectionProvider;
-import net.osmand.data.BackgroundType;
 import net.osmand.data.NearbyPlacePoint;
 import net.osmand.data.PointDescription;
 import net.osmand.plus.OsmandApplication;
-import net.osmand.plus.nearbyplaces.NearbyPlacesHelper;
-import net.osmand.plus.utils.AndroidUtils;
 import net.osmand.plus.utils.NativeUtilities;
 import net.osmand.plus.views.PointImageUtils;
-import net.osmand.plus.views.layers.NearbyPlacesLayer;
 import net.osmand.util.MapUtils;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
-import java.util.concurrent.ConcurrentHashMap;
 
 public class NearbyPlacesTileProvider extends interface_MapTiledCollectionProvider {
 
 	private static final Log log = LogFactory.getLog(NearbyPlacesTileProvider.class);
 	private final QListPointI points31 = new QListPointI();
 	private final List<MapLayerData> mapLayerDataList = new ArrayList<>();
-	private final Map<String, Bitmap> bigBitmapCache = new ConcurrentHashMap<>();
 	private Bitmap cachedSmallBitmap;
 	private final OsmandApplication app;
 	private final float density;
 	private final PointI offset;
 	private MapTiledCollectionProvider providerInstance;
 	private int baseOrder;
-	private final NearbyPlacesLayer nearbyPlacesLayer;
+	private Object selectedObject;
 
-
-	public NearbyPlacesTileProvider(@NonNull OsmandApplication context, NearbyPlacesLayer nearbyPlacesLayer, int baseOrder, float density) {
+	public NearbyPlacesTileProvider(@NonNull OsmandApplication context, int baseOrder, float density, Object selectedObject) {
```

Comment:
Recreating provider on object selection will cause flickering I guess. Probably there should be extra provider for selected object (like in `ContextMenuLayer`)